PROV-970 - Login form redirects to original dest

kehh opened the pull request.

kehh committed:
PROV-970 - Login form redirects to original dest

leftclickben commented on app/controllers/system/AuthController.php:
```diff
@@ -36,12 +36,21 @@ public function Login() {
+			// Redirect to the default action
+			$vs_redirect = caNavUrl($this->opo_request, null, null, null, null);
+			if(
+				isset($_SERVER['HTTP_REFERER']) &&
```
Not all servers or browsers support referrer, and it can be messed with by proxies.  Pass the redirect as an explicit HTTP GET parameter, i.e. /url/of/login/page?redirect=/url/of/page/we/need/to/login/for, when doing the original redirect.

kehh committed:
Feedback from review comments
* Add a redirect query parameter when getting kicked back to the login page
* Hide the redirect parameter for the login form
* fix whitespace

kehh commented:
I have updated the PR to reflect comments

leftclickben committed:
Don't call render() after a POST, always use a redirect().

+ After a failed login, instead of just rendering the "login" view on the POST URL
  for DoLogin, we need to redirect to the Login URL, passing through the redirect
  parameter.
+ After a successful login, instead of just rendering the "welcome" view on the
  POST URL for DoLogin, we need to redirect to either the passed-in redirect URL,
  or to the home page / dashboard URL.

leftclickben commented:
Updated to do proper redirects after a login, instead of just rendering a view on the response of a HTTP POST, which is almost never a good idea.
